check/configuration/supported/detekt.md

```diff
@@ -18,12 +18,34 @@ layout:
 
 [**Detekt**](https://github.com/detekt/detekt) is a linter for Kotlin.
 
-You can enable the Detekt linter with:
+
+detekt is composed of several linter commands.
```

Comment:
detekt is a bit more opinionated and a bit of a weird case. For detekt, basically it's the following:

- `detekt` is what most users will prefer, since it takes everything in detekt's default built-in config, and adds whatever you want in `.detekt.yaml`
- `detekt-explicit` is what some power users will prefer, where it omits detekt's default config, and uses `.detekt.yaml` as the source of truth. Only use this if you want to manually fine tune your `.detekt.yaml` to be whatever you want.
- `detekt-gradle` is if you are using gradle for all of your kotlin needs, and you already have detekt setup to run through gradle.

To make a food analogy,
- `detekt` is a premade meal kit that comes with a best effort set of rules
- `detekt-explicit` is the raw ingredients, and you have to tune everything yourself
- `detekt-gradle` is ordering in--it works and you can customize your order but you're at the mercy of gradle and their pipelines

I think this page will probably have to be populated manually. I'd recommend making `trunk check enable detekt` the first-class support, and then having a section on "other ways to use detekt", which goes over `detekt-explicit` and `detekt-gradle`